src/clj_kondo/impl/linters.clj

```diff
@@ -180,8 +180,8 @@
       (lint-missing-test-assertion ctx call))))
 
 (defn lint-arg-types! [ctx idacs call called-fn]
-  (when-let [arg-types (:arg-types call)]
-    (let [arg-types @arg-types
+  (when-let [args (and (:lint-arg-types? call) (:args call))]
```

Comment:
This can probably be reversed or is it related to this PR?

Reply:
it's actually related, now we always include the `:args` but it's not aways we want to lint the arg types, so we have a new boolean in ctx `lint-arg-types?` which we assoc and dissoc on places we don't want to lint (like master used to do dissocing the `arg-types`)